### PR TITLE
Data/fetch calls

### PR DESCRIPTION
**Pull request by jphorner**
*If a section is unnecessary, leave as (default)*

## What (if any) features are you implementing?

 - Added fetch calls for all movies and single movie
 - Added CSS to resize movie posters & display information

## What (if anything) did you refactor?

 - Adjusted error handling so that if a fetch call fails on individual movie display, the error message will display on that page
 - Removed pseudocode except where needed for future adjustments (MovieDetails)
 - Added conditional logic in render function (App) so that error messages would not persist when returning to main

## Were there any issues that arose?

 - (default)

## Is there anything needed from your teammates?

 - (default)

## Next intended features:

 - Will polish main page CSS & add sidebars + header

### CODE REVIEW

**Review by pthan1 — COMMENTED** [review]
We should figure out how to move the fetch calls into another file but this is great!

File: src/App.js [hunk 1]
```diff
@@ -23,56 +20,47 @@ class App extends Component {
...
-	hideSelectedMovieDetails = () => {
-		this.setState({ selectedMovie: {} });
+		fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+			.then((response) => {
+				if (response.ok) {
+					return response.json();
+				}
+				throw new Error('[all movies] Something went wrong. Please try again.');
+			})
+			.then((data) => {
```
Comment (kaylagordon) [inline src/App.js:30]: Do you use all of the data from the fetch? If not, clean up that data to ONLY what you need before setting it to state.

File: src/App.js [hunk 2]
```diff
@@ -23,56 +20,47 @@ class App extends Component {
...
+		fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/mmovies/${id}`)
+			.then((response) => {
+				if (response.ok) {
+					return response.json();
+				}
+				throw new Error('[for individual movie] Something went wrong. Please try again.');
+			})
+			.then((data) => {
+				if (!this.state.error) {
+					this.setState({ selectedMovie: data.movie });
```
Comment (kaylagordon) [inline src/App.js:47]: Same here - the fetch call gives you LOTS of data - if you're not using it, trim it down before setting to state.

File: src/App.js [hunk 3]
```diff
@@ -23,56 +20,47 @@ class App extends Component {
...
 	};
 
 	render() {
 		return (
 			<main>
 				<section className="app-body">
+					<aside className="left-sidebar" />
 					<Header />
+					{this.state.error && <h2>{this.state.error}</h2>}
 					{!this.state.isAMovieSelected ? (
```
Comment (kaylagordon) [inline src/App.js:64]: You are going to be able to delete `isMovieSelected` once you implement router!

File: src/App.js [hunk 4]
```diff
@@ -23,56 +20,47 @@ class App extends Component {
...
-
-	// showSelectedMovieDetails = (id) => {
-	// 	console.log(id);
-	// 	// Should hide banner and cards container
-	// 	// Should show movie details display
-	// };
-
-	hideSelectedMovieDetails = () => {
-		this.setState({ selectedMovie: {} });
+		fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
```
Comment (kaylagordon) [inline src/App.js:23]: Consider pulling your API calls into a separate file just like you did in mods 1 + 2.

**Pull request merged**